src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs

```diff
@@ -32,37 +31,21 @@ public IScriptingScope CreateScope(IEnumerable<GlobalMethod> methods, IServicePr
 
         public object Evaluate(IScriptingScope scope, string script)
         {
-            if (scope == null)
+            static void ThrowInvalidScopeTypeException()
             {
-                throw new ArgumentNullException(nameof(scope));
+                throw new ArgumentException($"Expected a scope of type {nameof(JavaScriptScope)}", nameof(scope));
```

Comment:
```suggestion
                ArgumentNullException.ThrowIfNull(scope, nameof(scope));
```

Reply:
Cannot throw null exception as this is case of wrong type of argument. Misleading the same way as FileNotFoundException would be.

Reply:
What I'm suggested is the same of what you wrote but with the new static version, feel free to accept it or reject it